doc/doc_guide.rst

```diff
@@ -0,0 +1,201 @@
+Writing documentation
+---------------------
+
+Marian’s documentation is generated using `Sphinx`_ + `Breathe`_ + `Doxygen`_ + `Exhale`_.
+`Doxygen`_ is used for documenting the source code and `Sphinx`_ (together with the extensions of
+`Breathe`_ and `Exhale`_) for managing handwritten documentation and generating library API
+reference.
+
+Whenever you add new code or propose changes to Marian, we would highly appreciate if you also add
+new Doxygen comments or update existing ones as needed along with your changes (see the `Doxygen
+guidelines`_ below). Your Doxygen comments will be integrated in the Marian’s documentation
+automatically.
+
+There is an ongoing and incremental effort with the goal of documenting essential Marian API in a
+consistent way. The existing code might not follow these guidelines, but new code should.
+
+
+Documentation with Doxygen
+``````````````````````````
+
+`Doxygen`_ is a powerful documentation system for C++ and many other languages that parses and
+extracts documentation comments included in the source code to generate a comprehensive
+documentation, for example, in HTML or LaTeX format.
+
+| **How to document your code using Doxygen?**
+| Doxygen recognises several special comment blocks with some additional markings. In Marian, we
```

Comment:
Is this supposed to generate a box or something similar? In my local setup Sphinx generates a normal text from it. Is it correct? Maybe it should be a subsection?